### PR TITLE
Credit problem

### PR DESCRIPTION
**Pull request by ramsha07**
Credit problem

### CODE REVIEW

File: week1/credit/credit.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#include <cs50.h>
+#include <stdio.h>
+
+int main(void)
+{
+    long number = 1234;
```
Comment (syedqasimahmed) [inline week1/credit/credit.c]: cardNumber

File: week1/credit/credit.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    int sum;
+    int digit;
+    int multiply1;
```
Comment (syedqasimahmed) [inline week1/credit/credit.c]: remove this

File: week1/credit/credit.c [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    int multiply;
```
Comment (syedqasimahmed) [inline week1/credit/credit.c]: remove this

File: week1/credit/credit.c [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    int multiply2;
```
Comment (syedqasimahmed) [inline week1/credit/credit.c]: remove this

File: week1/credit/credit.c [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    //         printf ("%i\n", sum);
+    //     }
+    //     else
+    //     {
+    //           sum = multiply+ digit1;
+    //     sum =+sum;
+    //     }
+        number = number / 10;//123/10 =12
+
+    }
```
Comment (syedqasimahmed) [inline week1/credit/credit.c]: Modify Logic
```javascript 
while (number > 0) {
int firstDigit = number % 10;// 4 
number = number / 10; // 123
int secondDigit = number % 10  *2 ; 

// Logic for second Digit 
sum = sum + secondDigit % 10; 
secondDigit = secondDigit / 10 ; 
sum = sum + secondDigit % 10;  

//  Add firstDigit also in the sum 
sum = sum + firstDigit ;

// In the last line divide number again

number = number /10;

}

```

**Comment by ramsha07** [comment]
Please check the updated version

File: week1/credit/credit.c [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        //  Add firstDigit also in the sum
+        sum = sum + firstDigit;
+        // In the last line divide number again
+        cardNumber = cardNumber / 10;
+    }
+    printf("%d \n", sum);
+
+    if (sum % 10 == 0)
+    {
+        printf("Valid\n");
```
Comment (syedqasimahmed) [inline week1/credit/credit.c]: Un necessory printf statement

File: week1/credit/credit.c [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+        while (cardNumber > 0)
+        {
+            int firstTwodigit = 0;
```
Comment (syedqasimahmed) [inline week1/credit/credit.c]: It should be camel case `firstTwoDigit`

File: week1/credit/credit.c [hunk 8]
```diff
@@ -0,0 +1,54 @@
+#include <cs50.h>
+#include <stdio.h>
+
+int main(void)
+{
+    long cardNumber = 378282246310005;
```
Comment (syedqasimahmed) [inline week1/credit/credit.c]: replace it and take user Input

File: week1/credit/credit.c [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+                }
+                else if (firstTwodigit == 34 || firstTwodigit == 37)
+                {
+                    printf("master\n");
+                }
+            }
+            cardNumber = cardNumber / 10;
+        }
+    }
+    else
```
Comment (syedqasimahmed) [inline week1/credit/credit.c:54]: There is common practice if return 
```javascript
if (sum % 10 != 0) {
printf("Valid\n");
return 0;
}
// rest of the code 
while (cardNumber > 0)
        { 
        ....
        }
```

File: week1/credit/credit.c [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    int sum = 0;
+    while (cardNumber > 0)
+    {
+        int firstDigit = cardNumber % 10; // 4
+        cardNumber = cardNumber / 10;     // 123
+        int secondDigit = cardNumber % 10 * 2;
+        // Logic for second Digit
+        sum = sum + secondDigit % 10;
+        secondDigit = secondDigit / 10;
+        sum = sum + secondDigit % 10;
```
Comment (syedqasimahmed) [inline week1/credit/credit.c]: these two lines modified by this `sum = sum + secondDigit / 10;`
so we have a cleaner version of this

File: week1/credit/credit.c [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            int firstTwodigit = 0;
+            if (cardNumber < 100)
+            {
+                firstTwodigit = cardNumber % 100;
+                if (firstTwodigit % 10 == 4)
+                {
+                    printf("Visa\n");
+                }
+                else if ((firstTwodigit == 51) || (firstTwodigit == 52) || (firstTwodigit == 53) || (firstTwodigit == 54) || (firstTwodigit == 55))
```
Comment (syedqasimahmed) [inline week1/credit/credit.c]: this condition could be change by `    else if (firstTwoDigits >= 51 && firstTwoDigits <= 55)`

**Review by syedqasimahmed — CHANGES_REQUESTED** [review]

**Comment by syedqasimahmed** [comment]
@ramsha07 Looks like the direction we want!

File: week1/credit/credit.c (hunk 6 above)
Comment (ramsha07) [reply week1/credit/credit.c]: i know it is unnecessary, it is just to confirm valid or invalid

File: week1/credit/credit.c [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+        // In the last line divide number again
+        cardNumber = cardNumber / 10;
+    }
+
+    if (sum % 10 != 0)
+    {
+        printf("INVALID\n");
+        return 0;
+    }
+    else
```
Comment (ramsha07) [inline week1/credit/credit.c]: remove else

File: week1/credit/credit.c [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+        while (number > 0)
+        {
+            int firstTwoDigit = 0;
```
Comment (ramsha07) [inline week1/credit/credit.c]: Define this with other variables. i.e (head of main)

File: week1/credit/credit.c [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+                {
+                    printf("AMEX\n");
+                }
+                else if (firstTwoDigit == 34 || firstTwoDigit == 37)
+                {
+                    printf("MASTERCARD\n");
+                }
+            }
+            number = number / 10;
+        }
```
Comment (ramsha07) [inline week1/credit/credit.c]: Modify logic 
while loop only resposible for counting `cardLength` and `firstTwoDigit`. Rest of the logic should be move out from while loop.

Another thing is compare card length as well with the starting digit.

File: week1/credit/credit.c [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+                    printf("VISA\n");
+                }
+                else if ((firstTwoDigit >= 51) && (firstTwoDigit <= 55))
+                {
+                    printf("AMEX\n");
+                }
+                else if (firstTwoDigit == 34 || firstTwoDigit == 37)
+                {
+                    printf("MASTERCARD\n");
+                }
```
Comment (ramsha07) [inline week1/credit/credit.c]: ADD one else with this if-else chain which prints INVALID

File: week1/credit/credit.c [hunk 16]
```diff
@@ -0,0 +1,50 @@
+#include <cs50.h>
+#include <stdio.h>
+
+int main(void)
+{
+    long cardNumber = get_long("Number: ");
+    long number = cardNumber;
```
Comment (ramsha07) [inline week1/credit/credit.c:7]: variable names should make sense.  (Optional)

File: week1/credit/credit.c [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+    int sum = 0;
```
Comment (ramsha07) [inline week1/credit/credit.c:8]: This variable name should be changed to `sumOfCardDigits` or `total` think a better name for this. (Optional)

File: week1/credit/credit.c [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+        {
+            int firstTwoDigit = 0;
+            if (number < 100)
+            {
+                firstTwoDigit = number % 100;
+                if (firstTwoDigit / 10 == 4)
+                {
+                    printf("VISA\n");
+                }
+                else if ((firstTwoDigit >= 51) && (firstTwoDigit <= 55))
```
Comment (ramsha07) [inline week1/credit/credit.c]: un necessory brackets should be removed and looks like this.

`else if (firstTwoDigit >= 51 && firstTwoDigit <= 55)`

File: week1/credit/credit.c [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+    long number = cardNumber;
+    int sum = 0;
+    while (cardNumber > 0)
+    {
+        int firstDigit = cardNumber % 10;
+        cardNumber = cardNumber / 10;
+        int secondDigit = cardNumber % 10 * 2;
+        // Logic for second Digit
+        sum = sum + secondDigit % 10;
+        sum = sum + secondDigit / 10;
```
Comment (ramsha07) [inline week1/credit/credit.c]: these two line can be simplified as 
```javascript
    sum = sum + secondDigit % 10  + secondDigit / 10; 
    ```

**Review by syedqasimahmed — APPROVED** [review]

**Comment by syedqasimahmed** [comment]
@ramsha07  Could you please rename the PR title and remove WIP: also merge this PR.

**Pull request merged**